src/ui/components/search/filtersearchcomponent.js

```diff
@@ -89,27 +90,42 @@ export default class FilterSearchComponent extends Component {
      * Optionally provided
      * @type {string}
      */
-    this.query = config.query || this.core.globalStorage.getState(`${StorageKeys.QUERY}.${this.name}`) || '';
-    this.core.globalStorage.on('update', `${StorageKeys.QUERY}.${this.name}`, q => {
+    this.query = config.query || this.core.globalStorage.getState(`${this.name}.${StorageKeys.QUERY}`) || '';
+    this.core.globalStorage.on('update', `${this.name}.${StorageKeys.QUERY}`, q => {
       this.query = q;
       this.search();
     });
 
     /**
      * The filter string to use for the provided query
      * Optionally provided
+     * TODO(oshi): config.filter is not in the readme. Do we need this?
      * @type {string}
      */
-    this.filter = config.filter || this.core.globalStorage.getState(`${StorageKeys.FILTER}.${this.name}`) || '';
-    if (typeof this.filter === 'string') {
+    let filter = config.filter || '';
+    if (typeof filter === 'string') {
       try {
-        this.filter = JSON.parse(this.filter);
+        filter = JSON.parse(filter);
       } catch (e) {}
     }
 
-    this.searchParameters = buildSearchParameters(config.searchParameters);
+    let filterNode = this.core.globalStorage.getState(`${this.name}.${StorageKeys.STATIC_FILTER_NODE}`) || {};
+    if (typeof filterNode === 'string') {
+      try {
+        filterNode = JSON.parse(filterNode);
+      } catch (e) {}
+    }
 
-    this.core.globalStorage.on('update', `${StorageKeys.FILTER}.${this.name}`, f => { this.filter = f; });
+    if (filter) {
+      this.filterNode = FilterNodeFactory.from({
+        ...filterNode,
```

Comment:
Tried this and the filternode will be correctly created (seen through console logging), so there's no functional issues with this. I think I can remove setting this.filterNode in the constructor, though, because in order to store the filternode the autocomplete has to be submitted, and when that happens a new filternode is generated.